### PR TITLE
Adds the basic functionality for chat and socket io

### PR DESCRIPTION
**Pull request by aysam199**
changed the server creation from server.js to index.js (not sure if we can delete server.js yet because of heroku), added app.js and the socket io functionality.

### CODE REVIEW

File: server/package.json [hunk 1]
```diff
@@ -5,7 +5,7 @@
   "main": "index.js",
   "scripts": {
     "test": "echo \"Error: no test specified\" && exit 1",
-    "start": "npm i && node server.js"
+    "start": "node ./app/routes/index.js"
```
Comment (Karyum) [inline server/package.json]: i dont understand why make an `app.js` file and then use a different file to run the server ?

Comment (aysam199) [reply server/package.json]: the app.js is not used to run the server, its used to deal with static files and so on

Comment (Karyum) [reply server/package.json]: oh my bad then

Comment (Karyum) [reply server/package.json]: @aysam199 but why are we using handlebars ? instead of making it work with React ?

Comment (aysam199) [reply server/package.json]: I'm a bit confused, the handlebars and express are for the backend, i make a request from the front end with react to be handled by the backend handlebars. I cant make it just work with react i need it on the backend aswell

Comment (Karyum) [reply server/package.json]: yes you need the backend, for the data. but the react is the one that handles the frontend.
handlebars also does the same as react, so we can't have 2 frontend frameworks, we need only 1

File: server/app/routes/GroupChat/GroupChat.html [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  <body>
+    <ul id="messages"></ul>
+    <form action="">
+      <input id="m" autocomplete="off" /><button>Send</button>
+    </form>
+    
+    <script src="/socket.io/socket.io.js"></script>
+    <script src="GroupChat.js"></script>
+  </body>
+</html>
```
Comment (Karyum) [inline server/app/routes/GroupChat/GroupChat.html]: now we need to delete this file

File: server/app/routes/GroupChat/GroupChat.js [hunk 3]
```diff
@@ -0,0 +1,3 @@
+document.addEventListener("DOMContentLoaded", function() {
+var socket = io();
```
Comment (Karyum) [inline server/app/routes/GroupChat/GroupChat.js]: and this file cause we dont need them anymore

File: server/app/routes/GroupChat/GroupChat.html (hunk 2 above)
Comment (aysam199) [reply server/app/routes/GroupChat/GroupChat.html]: We need the form for chatting, did you mean to put it in the front end?

File: server/app/routes/GroupChat/GroupChat.js (hunk 3 above)
Comment (aysam199) [reply server/app/routes/GroupChat/GroupChat.js]: This one fixed an error with recognizing the io(), but essentially we dont need it.

File: server/app/routes/GroupChat/GroupChat.html (hunk 2 above)
Comment (Karyum) [reply server/app/routes/GroupChat/GroupChat.html]: or yeah move it to the frontend that is also an option

File: server/app/routes/GroupChat/GroupChat.js (hunk 3 above)
Comment (Karyum) [reply server/app/routes/GroupChat/GroupChat.js]: ok to the frontend then

**Comment by Karyum** [comment]
@aysam199 whats the progress of this PR?

File: client/public/index.html [hunk 4]
```diff
@@ -23,5 +23,11 @@
   </head>
   <body>
     <div id="root"></div>
+    <script src="https://cdnjs.cloudflare.com/ajax/libs/socket.io/2.3.0/socket.io.js"></script>
+        <script>
+          document.addEventListener("DOMContentLoaded", function() {
+          var socket = io.connect("http://localhost:5000/")
```
Comment (Karyum) [inline client/public/index.html]: ahh we can't do this like that, it wouldn't work on Netlify

File: client/src/components/GroupChat/GroupChat.css [hunk 5]
```diff
@@ -0,0 +1,8 @@
+* { margin: 0; padding: 0; box-sizing: border-box; }
+body { font: 13px Helvetica, Arial; }
+form { background: #000; padding: 3px; position: fixed; bottom: 0; width: 100%; }
+form input { border: 0; padding: 10px; width: 90%; margin-right: .5%; }
+form button { width: 9%; background: rgb(130, 224, 255); border: none; padding: 10px; }
+#messages { list-style-type: none; margin: 0; padding: 0; }
+#messages li { padding: 5px 10px; }
+#messages li:nth-child(odd) { background: #eee; }
```
Comment (Karyum) [inline client/src/components/GroupChat/GroupChat.css]: i feel like this code is copy pasted, atleast format it :sweat_smile:

File: client/src/components/SelectedCourseGroups/SelectedCourseGroups.js [hunk 6]
```diff
@@ -43,7 +46,9 @@ const classes = useStyles();
 
         {tileData.map(tile => (
           <GridListTile key={tile.img}>
-            <img src={tile.img} alt={tile.title} />
+            <Link className='img-link' to={'/GroupChat'}>
```
Comment (Karyum) [inline client/src/components/SelectedCourseGroups/SelectedCourseGroups.js]:
```suggestion
            <Link className='img-link' to='/GroupChat'>
```

File: client/src/components/GroupChat/GroupChat.css (hunk 5 above)
Comment (aysam199) [reply client/src/components/GroupChat/GroupChat.css]: its from their website, the design comes later we just want it working now.

File: client/public/index.html (hunk 4 above)
Comment (aysam199) [reply client/public/index.html]: why wouldnt it? how should it be?

**Comment by Karyum** [comment]
@aysam199 why did you close it?

**Pull request closed without merging**